torch_xla/experimental/distributed_checkpoint/manager.py

```diff
@@ -105,6 +109,27 @@ def __init__(self,
     self.save_period = save_period
     self.max_to_keep = max_to_keep
 
+    self._async_queue = queue.Queue(maxsize=async_queue_size)
+    self._chkpt_thread = threading.Thread(target=self._async_worker, daemon=True)
+    self._chkpt_thread.start()
+
+    # Create a CPU process group to coordinate the checkpoint.
+    self.pg = dist.new_group(backend='gloo')
+
+  def __del__(self):
+    # Ensure pending checkpoints are finished
+    self._async_queue.join()
```

Comment:
Interesting ideas!

Reply:
Ah unfortunately it seems I can't count on __del__ to be called... I'll have to revisit this.

Reply:
Then, you may also want to avoid spawning `daemon` threads.

Reply:
I have a better understanding now - I can't rely on it immediately after calling `del chkpt_mgr` like I was in the unit tests, but it will be called before the process exits.